src/InstantiableEntity.php

```diff
@@ -8,8 +8,12 @@
 
 class InstantiableEntity extends BaseEntity {
 
-	function __construct( $name, array $properties = [] ) {
-		$this->name = $name;
+	protected $name;
+
+	function __construct( $name = null, array $properties = [] ) {
+		if ( isset( $name ) ) {
```

Comment:
Side note - I recently learned that the original meaning of the word "moot" was "subject to debate", which makes the true meaning of "moot" moot (i.e. debatable) if you are pedantic, but that argument is moot (i.e. irrelevant) considering the widely accepted meaning these days: "irrelevant". How meta!